plugins/outputs/opensearch/opensearch.go

```diff
@@ -0,0 +1,451 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package opensearch
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchutil"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal/choice"
+	httpconfig "github.com/influxdata/telegraf/plugins/common/http"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Opensearch struct {
+	AuthBearerToken     string          `toml:"auth_bearer_token"`
+	DefaultPipeline     string          `toml:"default_pipeline"`
+	DefaultTagValue     string          `toml:"default_tag_value"`
+	EnableGzip          bool            `toml:"enable_gzip"`
+	EnableSniffer       bool            `toml:"enable_sniffer"`
+	FloatHandling       string          `toml:"float_handling"`
+	FloatReplacement    float64         `toml:"float_replacement_value"`
+	ForceDocumentID     bool            `toml:"force_document_id"`
+	HealthCheckInterval config.Duration `toml:"health_check_interval"`
+	HealthCheckTimeout  config.Duration `toml:"health_check_timeout"`
+	IndexName           string          `toml:"index_name"`
+	ManageTemplate      bool            `toml:"manage_template"`
+	OverwriteTemplate   bool            `toml:"overwrite_template"`
+	Password            config.Secret   `toml:"password"`
+	TemplateName        string          `toml:"template_name"`
+	Timeout             config.Duration `toml:"timeout"`
+	URLs                []string        `toml:"urls"`
+	UsePipeline         string          `toml:"use_pipeline"`
+	Username            config.Secret   `toml:"username"`
+	Log                 telegraf.Logger `toml:"-"`
+	pipelineName        string
+	pipelineTagKeys     []string
+	tagKeys             []string
+	onSucc              func(context.Context, opensearchutil.BulkIndexerItem, opensearchutil.BulkIndexerResponseItem)
+	onFail              func(context.Context, opensearchutil.BulkIndexerItem, opensearchutil.BulkIndexerResponseItem, error)
+	httpconfig.HTTPClientConfig
+	osClient *opensearch.Client
+}
+
+//go:embed telegrafTemplate.json
+var telegrafTemplate string
+
+type templatePart struct {
+	TemplatePattern string
+}
+
+func (*Opensearch) SampleConfig() string {
+	return sampleConfig
+}
+
+func (o *Opensearch) Init() error {
+	if o.URLs == nil || o.IndexName == "" {
```

Comment:
```suggestion
	if len(o.URLs) == 0 || o.IndexName == "" {
```

Maybe you should also check if the value in each URL is not and empty string (as in your current `sample.conf`) to exit early.

Reply:
done